src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakController.kt

```diff
@@ -62,4 +63,13 @@ class VedtakController(private val stegService: StegService,
         tilgangService.validerTilgangTilBehandling(behandlingId)
         return Ressurs.success(vedtakService.hentVedtakHvisEksisterer(behandlingId))
     }
+
+    @GetMapping("/eksternid/{eksternId}")
+    fun hentVedtakForEksternId(@PathVariable eksternId: Long, dato: LocalDate = LocalDate.now()): Ressurs<Int> {
+        val behandlingId = behandlingService.hentBehandlingPåEksternId(eksternId).id
+
+        val beregnetInntekt = vedtakService.hentForventetInntektForVedtakOgDato(behandlingId, dato)
+                              ?: return Ressurs.funksjonellFeil("Fant ingen forventet inntekt for behandling $behandlingId og gitt dato $dato")
+        return Ressurs.success(beregnetInntekt)
```

Comment:
Hvis man ikke får innvilget så har man ikke noen inntekt, då kommer denne kaste funksjoellt feil.
Mulig man burde wrappa responset i et eget objekt som sier om den har inntekt (innvilget/avslått?)

Reply:
Personhendelse kommer antakelig bare til å kalle på personer som har fått innvilget.
Hvis jeg ikke husker helt feil så vil avslåtte ikke få efvedtak-hendelser, og dermed ikke bli kalt av personhendelse.
Dersom noen får innvilget så burde man kunne forvente at det er registrert beregnet inntekt går jeg ut ifra, ellers kan det kastes funksjonell feil.
Mulig det er behov for å vite om en person ikke har inntekt som følge av at vedkommende fikk avslått senere, men akkurat nå ser jeg ikke behovet for å vite om det er for avslått eller ikke.
Jeg må nok uansett lage noe som sjekker om en person har fortsatt løpende vedtak eget sted i personhendelse.

Reply:
👍

Reply:
Sant, finnes nog ingen grunn till att vi skal sende melding om fattet vedtak når det blitt avslått heller, så hvis det skulle være sånn burde vi vurdere om vi trenger det :)